Code/Tools/SceneAPI/SceneCore/Events/AssetImportRequest.cpp

```diff
@@ -17,13 +17,103 @@
 #include <SceneAPI/SceneCore/Events/AssetImportRequest.h>
 #include <SceneAPI/SceneCore/Events/SceneSerializationBus.h>
 #include <SceneAPI/SceneCore/Utilities/Reporting.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzToolsFramework/Debug/TraceContext.h>
+#include <AzCore/JSON/writer.h>
+#include <AzCore/IO/TextStreamWriters.h>
 
 namespace AZ
 {
     namespace SceneAPI
     {
         namespace Events
         {
+            struct UpdateManifestScope final
+                : public AssetImportRequestReporter
+            {
+                AZ_RTTI(UpdateManifestScope, "{C9DD605E-2A9D-4C0D-A22E-A71B44528420}", AssetImportRequestReporter);
+
+                UpdateManifestScope(AZStd::shared_ptr<Containers::Scene> scene)
+                    : m_scene(scene)
+                {
+                    m_registered = false;
+                    AZ::SettingsRegistry::Get()->Get(m_registered, AZ::SceneAPI::Utilities::Key_AssetProcessorInDebug);
+                    if (m_registered)
+                    {
+                        AZ::Interface<AssetImportRequestReporter>::Register(this);
+                    }
+                }
+
+                ~UpdateManifestScope()
+                {
+                    if (m_registered)
+                    {
+                        AZ::Interface<AssetImportRequestReporter>::Unregister(this);
+                        m_registered = false;
+                    }
+                    m_previous = {};
+                    m_previousSet = false;
+                }
+
+                void ReportStart([[maybe_unused]] const AssetImportRequest* instance) override
+                {
+                    ++m_reportNumber;
+
+                    if (!m_previousSet)
+                    {
+                        auto saveToJsonOutcome = m_scene->GetManifest().SaveToJsonDocument();
+                        if (!saveToJsonOutcome.IsSuccess())
+                        {
+                            AZ_Error(AZ::SceneAPI::Utilities::ErrorWindow, false,
```

Comment:
Would this make more sense as a warning? If the manifest debug file fails to save, the product assets should still be fine and the rest of processing should be good, right?

Reply:
I guess... the SaveToJsonDocument() would fail if an abstract class inside the scene manifest is not registered by the Serial Context. This means the code will need to be updated. Based on the "AZ_Error()" vs "AZ_Warning()" choice this seems to be a code fault requiring a code fix.. so therefore a "AZ_Error()" right?

Reply:
Sure, I think that's fine for now.

I think we're still planning out how we want to better separate error/warning messages between "This is a content issue, you can fix it by changing content" and "This is a code issue, and code needs to be changed to address it"